### DISCUSSION TITLE
v1.5: Demo Modules (breaking change)

### DISCUSSION THREAD

**Discussion by puckpuck** [open-telemetry/opentelemetry-demo#927] [Announcements]
The demo has grown significantly in size and complexity, which is impacting the getting started experience in negative ways. A decision was made to split the demo into its core components and several optional modules. The default mode to run the demo will be the core components only, with options to start one or more modules. The #872 PR contains these changes.

The core components will only have the core services required to start the demo with a browser-accessible UI. It will include an OpenTelemetry Collector, but the Collector will not send data to any telemetry backend in this mode.

The optional modules will be:

- `load` - includes the load generator
- `observability` - includes jaeger, prometheus, and grafana, as well as necessary changes to the collector
- `featureflags` - includes the feature flag service with postrgres and an environment variable flag for several services that have logic for feature flags.
- `kafka-extras` - includes kafka, accounting, and fraud detection services as well as an environment variable flag for the checkout service
- `tests` - tests included with demo

A full version of the demo can be run with a special `full` or `all` module when specified with the start script.

We are still working through some of the details on how to start and stop the demo compared to how it is done today using `docker compose up|down` commands to ensure users do not get errors or unpleasant surprises with this change.

**Comment by julianocosta89** [comment]
Hey @puckpuck,.
Sorry I've missed the last sync calls and for sure that was discussed, but can't we make the default the full version?

If someone wants to run just part of the demo, they can specify which part they want, but IMO `docker compose up` should always start the whole demo.

**Comment by puckpuck** [comment]
The PR has been updated. The original goal was to make a minimal mode of the demo for people to run in resource-constrained environments. As such, we decided to forgo the complexity with "modules" and introduce just a minimal mode in which the demo can be run. The default will still be to run the entire full demo, so `docker compose up` will continue functioning as before.

In an effort to minimize complexity, we also will introduce make targets to start the demo in full or minimal mode, as well as a target to stop the demo.
